src/main/java/Entities/UserMatches/UserMatches.java

```diff
@@ -0,0 +1,77 @@
+package Entities.UserMatches;
+
+import Entities.Group;
+import Entities.User;
+import Entities.UserPublicProfile;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class UserMatches implements MatchingAlgorithm {
+    private User u;
+    private List<Group> matches;
+
+    private List<Group> groups;
+
+    public UserMatches(User user, List<Group> groups){
+        this.u = user;
+        this.matches = groups;
+        //Sort the matches
+        this.match();
+        this.groups = groups;
+    }
+
+
+    //Match the groups and Rank them!
+    @Override
+    public void match() {
```

Comment:
Instead of having the match method put the similarity score and and group into a hash map, as mentioned before we could have a list of matches, where match is it's own class that has a similarity score instance attribute (this can be defined in the constructor the same way that calculateSimilarity() method is defining the similarity score). This match could implement comparable and then we can get rid of this method and just sort the list of matches using Collections' sort method. I think this will follow the clean architecture principle of single responsibility better and also get rid of some methods.